draft-spencer-haapi/draft-spencer-haapi-protocol.md

```diff
@@ -0,0 +1,308 @@
+
+An authorization server (in the OAuth and OpenID Connect sense of the term) supporting this specification provides an REST API that authenticates users operating attested client applications. This interaction is shown in figure 1:
+
+```text
++--------------------+                                   Token   +--------------------+
+|                    |              CAT                  endpoint|                    |
+|                    +--(A)---------------------------->o--------+                    |
+|                    |                                           |                    |
+|                    |<-(B)--------------------------------------+                    |
+|                    |  API AT                                   |                    |
+|                    |                              Authorization|                    |
+|                    |   API AT & PoP                    endpoint|                    |
+|                    +--(C)---------------------------->o--------+                    |
+|                    |                                           |                    |
+|                    |                             Authentication|                    |
+|       OAuth        |   API AT & PoP              endpoint      |    Authorization   |
+|      Client        +--(D)---------------------------->o--------+       Server       |
+|    Application     |       Login / authentication of user      |                    |
+|                    |                                           |                    |
+|                    |<-(E)--------------------------------------+                    |
+|                    |                     LT                    |                    |
+|                    |                              Authorization|                    |
+|                    |       LT, API AT & PoP            endpoint|                    |
+|                    +--(F)---------------------------->o--------+                    |
+|                    |                                           |                    |
+|                    ---(G)--------------------------------------+                    |
+|                    |                Access Token               |                    |
++--------------------+                                           +--------------------+
+```
+
+> Figure 1: The integration of an OAuth client application with an authorization server 
+
+The steps a client makes to obtain an access token are thus:
+
+(A) The OAuth client application, in possession of a Client Attestation Token (CAT) obtained by the process depicted in figure 1 and 2 below, uses the OAuth "Client Credential Grant" as a representation of the appropriate consent. This CAT is a JSON Web Token (JWT) which is verified using the Assertion Framework for OAuth 2.0 Client Authentication and Authorization Grants.
+
+(B) The authorization server issues an API Access Token (AT).
```

Comment:
suggestion: The authorization server issues a HAAPI Access Token, which is a PoP token bound to a key that the client holds.

Reply:
Will add some info here about key binding. Good idea.